systemd/systemd_test.go

```diff
@@ -228,58 +240,168 @@ func (s *SystemdTestSuite) TestStartNoBlock(c *C) {
 	c.Check(s.argses, DeepEquals, [][]string{{"start", "--no-block", "foo", "bar"}})
 }
 
-func (s *SystemdTestSuite) TestStop(c *C) {
-	restore := MockStopDelays(time.Millisecond, 25*time.Second)
+// Stop failed because 'systemctl stop --no-block' returned an error
+func (s *SystemdTestSuite) TestStopSystemCtlStopError(c *C) {
+	// Poll rate = 1ms, Silence timeout = 2ms
+	restore := MockStopDelays(time.Millisecond, 2*time.Millisecond)
+	defer restore()
+	s.outs = [][]byte{
+		nil, // for the "stop" itself
+	}
+	s.errors = []error{&SystemctlError{"Mock Error"}}
+	// Timeout = 10ms
+	err := New(SystemMode, s.rep).Stop([]string{"foo"}, 10*time.Millisecond)
+	c.Assert(err, ErrorMatches, "Systemctl failed with error: Mock Error")
+	c.Assert(s.argses, HasLen, 1)
+	c.Check(s.argses[0], DeepEquals, []string{"stop", "--no-block", "foo"})
+	// No notifications
+	c.Assert(len(s.rep.msgs), Equals, 0)
+}
+
+// Stop failed because 'systemctl show' returned an error
+func (s *SystemdTestSuite) TestSystemCtlShowError(c *C) {
+	// Poll rate = 1ms, Silence timeout = 2ms
+	restore := MockStopDelays(time.Millisecond, 2*time.Millisecond)
+	defer restore()
+	s.outs = [][]byte{
+		nil, // for the "stop" itself
+		nil, // show
+	}
+	s.errors = []error{nil, &SystemctlError{"Mock Error"}}
```

Comment:
errors.New("mock error") here and other places

Reply:
Done.